docs/source/amp.rst

```diff
@@ -74,6 +74,15 @@ so they don't flush to zero.
 Each parameter's gradient (``.grad`` attribute) should be unscaled before the optimizer
 updates the parameters, so the scale factor does not interfere with the learning rate.
 
+.. note::
+
+  AMP/fp16 may not be for every model! For example, most bf16-pretrained models cannot operate in
+  the fp16 numerical range of max 65k and will cause gradients to overflow instead of underflow. In
+  this case, the scale factor may decrease under 1 as an attempt to bring gradients to a number
+  representable in fp16. While one may expect the scale to always be above 1, our GradScaler does
+  NOT make this guarantee to maintain performance. If you encounter NaNs in your loss or gradients
+  when running with AMP or fp16, verify your model is compatible.
```

Comment:
```suggestion
  AMP/fp16 may not work for every model! For example, most bf16-pretrained models cannot operate in
  the fp16 numerical range of max 64k and will cause gradients to overflow instead of underflow. In
  this case, the scale factor may decrease under 1 as an attempt to bring gradients to a number
  representable in fp16 dynamic range. While one may expect the scale to always be above 1, our GradScaler does
  NOT make this guarantee to maintain performance. If you encounter NaNs in your loss or gradients
  when running with AMP/fp16, verify your model is compatible.
```

Reply:
Thanks for the comments! I thought the max for fp16 is 65.5k or something, no?

Reply:
The maximum representable value is (2−2−10) × 215 = 65504. 

which is 64K (65504/2014)

Reply:
Ah, I see the distinction was regarding the K vs k. I'm just going to use the actual number for maximal clarity.